### ISSUE TITLE
Bug in encoder_layer?

### ISSUE THREAD

**Issue by jjjjohnson** [mobvoi/wenet#303]
Looks like the line blow is wrong when output_cache is not None, 
https://github.com/mobvoi/wenet/blob/19af50a6c848e77adb5dfa40a861061c1ae8b78e/wenet/transformer/encoder_layer.py#L232

should be : `x_concat = torch.cat((x_q, x_att), dim=-1)`, since x_att.shape[1] != x.shape[1] when output_cache is not None.

**Comment by whiteshirt0429** [comment]
this branch is not used, since we usually use add instead of concat to finish skip connection. Anyway, we will check it, thanks.

**Comment by jjjjohnson** [comment]
@whiteshirt0429 Thanks for your apply! Great work!
